sklearn/utils/estimator_checks.py

```diff
@@ -546,6 +548,12 @@ def checks_generator():
     )
 
 
+@validate_params(
+    {
+        "estimator": [BaseEstimator, None],
```

Comment:
We should not expect people to pass an estimator that inherits from `BaseEstimator`. Since the purpose of the class itself is to check the validity of the estimator, we don't need to validate the parameter using the string `"no_validation"`.

Reply:
I made the changes accordingly since they were minimal.